# Brazilian E-Commerce Public Dataset by Olist (On Going)

##### This project explores and cleans the Olist Brazilian E-Commerce dataset, a large real-world dataset containing customer, order, payment, product, and review information.
##### The goal is to prepare a clean, analysis-ready dataset to study key business questions such as:
##### - Customer behavior and satisfaction
##### - Payment methods and spending patterns
##### - Delivery performance and logistics
##### - Seller and product insights. 

## Steps:
### 1. Installing Kaggle and importing the data set
### 2. Import Libraries
### 3. Download all related data sets
### 4. Clean and explore all data sets before merging
#### - 4.1. Olist Customer Dataset
#### - 4.2. Olist Geolocation Dataset
#### - 4.3. Olist Order Items Dataset
#### - 4.4. Olist Order Payments Dataset
#### - 4.5. Olist Order Reviews Dataset
#### - 4.6. Olist Orders Dataset


## 1. Installing Kaggle and importing the data set

In [4]:
# Install kaggle API
!pip install Kaggle

In [5]:
# Verify Kaggle is installed
import kaggle
print ("Kaggle API is installed")

Kaggle API is installed


In [6]:
# list the datasets
!kaggle datasets list

ref                                                             title                                               size  lastUpdated          downloadCount  voteCount  usabilityRating  
--------------------------------------------------------------  -------------------------------------------------  -----  -------------------  -------------  ---------  ---------------  
ahmeduzaki/global-earthquake-tsunami-risk-assessment-dataset    Global Earthquake-Tsunami Risk Assessment Dataset   16KB  2025-10-01 16:35:53          17801        612  1.0              
jaderz/hospital-beds-management                                 Hospital Beds Management                            46KB  2025-10-03 09:21:58          15037        356  1.0              
jockeroika/life-style-data                                      Life Style Data                                      4MB  2025-10-14 13:50:45          20717        405  0.8235294        
rehan497/students-social-media-addiction                        S

In [7]:
!kaggle datasets list -s "olist"

ref                                                          title                                              size  lastUpdated          downloadCount  voteCount  usabilityRating  
-----------------------------------------------------------  ------------------------------------------------  -----  -------------------  -------------  ---------  ---------------  
olistbr/brazilian-ecommerce                                  Brazilian E-Commerce Public Dataset by Olist       43MB  2021-10-01 19:08:27         397397       3755  1.0              
olistbr/marketing-funnel-olist                               Marketing Funnel by Olist                         278KB  2018-11-16 14:00:20          17374        316  1.0              
terencicp/e-commerce-dataset-by-olist-as-an-sqlite-database  E-commerce dataset by Olist (SQLite)               49MB  2024-04-28 14:56:35           8765         89  1.0              
erak1006/brazilian-e-commerce-company-olist                  Brazilian e-commerce com

In [8]:
# download the dataset
!kaggle datasets download -d olistbr/brazilian-ecommerce --unzip

Dataset URL: https://www.kaggle.com/datasets/olistbr/brazilian-ecommerce
License(s): CC-BY-NC-SA-4.0
100%|██████████████████████████████████████| 42.6M/42.6M [00:43<00:00, 1.34MB/s]
100%|██████████████████████████████████████| 42.6M/42.6M [00:43<00:00, 1.02MB/s]


## 2. Import libraries

In [10]:
import os
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [11]:
# Checking the present files in the directory
print(os.listdir())

['.config', 'Music', '.kaggle', '.condarc', 'olist_sellers_dataset.csv', 'Untitled1.ipynb', '.DS_Store', '.dbvis', 'product_category_name_translation.csv', '.CFUserTextEncoding', 'Udemy-backup.ipynb', '.xonshrc', 'anaconda_projects', 'Full Disk Access', '.zshrc', 'olist_orders_dataset.csv', '.psql_history', 'Pictures', 'Udemy.ipynb', 'Entertainment', 'olist_order_items_dataset.csv', '.zsh_history', 'Untitled2.ipynb', '.ipython', 'Desktop', 'Library', '.matplotlib', '.pgadmin', 'Public', '.tcshrc', 'olist_customers_dataset.csv', '.virtual_documents', '.anaconda', 'Movies', 'Applications', 'udemy_courses.csv', 'olist_geolocation_dataset.csv', '.Trash', 'olist_order_payments_dataset.csv', '.ipynb_checkpoints', '.jupyter', 'Documents', '.vscode', '.bash_profile', 'Photos', 'Work-Related', 'Brazilian E-Commerce Public Dataset by Olist.ipynb', 'Downloads', '.continuum', '.zsh_sessions', 'olist_order_reviews_dataset.csv', '.conda', 'olist_products_dataset.csv']


## 3. Download all related data sets

In [232]:
# Read the CSV files
df_olist_customers = pd.read_csv("olist_customers_dataset.csv")
df_olist_geolocation = pd.read_csv("olist_geolocation_dataset.csv")
df_olist_order_items = pd.read_csv("olist_order_items_dataset.csv")
df_olist_order_payments = pd.read_csv("olist_order_payments_dataset.csv")
df_olist_order_reviews = pd.read_csv("olist_order_reviews_dataset.csv")
df_olist_orders = pd.read_csv("olist_orders_dataset.csv")
df_olist_products = pd.read_csv("olist_products_dataset.csv")
df_olist_sellers = pd.read_csv("olist_sellers_dataset.csv")
df_olist_product_category_name = pd.read_csv("product_category_name_translation.csv")

## 4. Clean and explore all data sets before merging

### 4.1. Olist Customer Dataset

In [241]:
# Read the CSV file
df_olist_customers.head()

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

##### customer_id seems to be a unique identifier for each order (same person can have several), which can be used as a key to join with the other tables.
##### customer_unique_id seems to represent the actual customer (unique per person) that can be used for customer-level analysis, e.g., “how many returning customers.”

In [15]:
df_olist_customers.shape

(99441, 5)

In [16]:
df_olist_customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [17]:
df_olist_customers.isna().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

In [18]:
# checking for duplicates (customer_id should appear only once)
df_olist_customers.duplicated(subset= 'customer_id').sum()

0

In [19]:
# validate ID formats (all IDs should have the same length and be alphanumeric (hex_style) )
df_olist_customers['customer_id'].str.len().value_counts().head()
df_olist_customers['customer_unique_id'].str.len().value_counts().head()

customer_unique_id
32    99441
Name: count, dtype: int64

##### The length is the expected uniform 32

In [21]:
# Normalize the text columns (cities and states should be consistent)
df_olist_customers['customer_city']= (df_olist_customers['customer_city'].str.lower().str.strip())
df_olist_customers['customer_state']= (df_olist_customers['customer_state'].str.lower().str.strip())

In [22]:
# check postal codes for unrealistic values (Brazilian ZIP prefixes usually range roughly between 01000–99999)
df_olist_customers['customer_zip_code_prefix'].describe()

count    99441.000000
mean     35137.474583
std      29797.938996
min       1003.000000
25%      11347.000000
50%      24416.000000
75%      58900.000000
max      99990.000000
Name: customer_zip_code_prefix, dtype: float64

In [23]:
# Remove special characters or whitespace in text (they might cause merging issues later)
df_olist_customers= df_olist_customers.applymap(
    lambda x: x.strip() if isinstance(x, str) else x
)

/var/folders/hq/_gnsfjyd47q9z3ll9llqlfpw0000gn/T/ipykernel_1631/2864326162.py:2: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  df_olist_customers= df_olist_customers.applymap(


### 4.2. Olist Geolocation Dataset

In [69]:
df_olist_geolocation.head()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                         1037       -23.545621       -46.639292   
1                         1046       -23.546081       -46.644820   
2                         1046       -23.546129       -46.642951   
3                         1041       -23.544392       -46.639499   
4                         1035       -23.541578       -46.641607   

  geolocation_city geolocation_state  
0        sao paulo                SP  
1        sao paulo                SP  
2        sao paulo                SP  
3        sao paulo                SP  
4        sao paulo                SP

In [65]:
df_olist_geolocation.shape

(1000163, 5)

##### Here are 1,000,163 rows, but not unique ZIP codes
##### That’s much larger than the customers dataset (99,441 rows).
##### This means each postal code appears many times with slightly different latitude and longitude values, because each postal prefix covers multiple locations (e.g., different buildings or streets in the same area).

In [76]:
df_olist_geolocation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  object 
 4   geolocation_state            1000163 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 38.2+ MB


##### No missing values at all, excellent for merging and mapping.
##### All data types make perfect sense.

In [81]:
df_olist_geolocation.duplicated().sum()

261831

##### The geolocation dataset is a reference table that maps ZIP code prefixes to geographic coordinates. It is not transactional data.
##### So, I decide to remove them

In [87]:
# Remove exact duplicates
df_olist_geolocation = df_olist_geolocation.drop_duplicates()

# Confirm result
print("Remaining duplicates:", df_olist_geolocation.duplicated().sum())
print("New shape:", df_olist_geolocation.shape)

Remaining duplicates: 0
New shape: (738332, 5)


In [97]:
# Standardize text columns
df_olist_geolocation.loc[:, 'geolocation_city'] = (
    df_olist_geolocation['geolocation_city'].str.strip().str.lower()
)

df_olist_geolocation.loc[:, 'geolocation_state'] = (
    df_olist_geolocation['geolocation_state'].str.strip().str.lower()
)

In [101]:
# validate latitude and longitude ranges
df_olist_geolocation[['geolocation_lat', 'geolocation_lng']].describe()

geolocation_lat  geolocation_lng
count    738332.000000    738332.000000
mean        -20.998353       -46.461098
std           5.892315         4.393705
min         -36.605374      -101.466766
25%         -23.603061       -48.867822
50%         -22.873588       -46.647278
75%         -19.923336       -43.836974
max          45.065933       121.105394

##### Brazil’s approximate bounding box:
##### Latitude: −35 ≤ lat ≤ +5
##### Longitude: −75 ≤ lng ≤ −30
##### Everything outside that is invalid for Brazil, they are clearly a few outliers, but I am curius to find out how many.

In [106]:
invalid_coords = df_olist_geolocation.query(
    "geolocation_lat < -35 or geolocation_lat > 5 or geolocation_lng < -75 or geolocation_lng > -30"
)

print("Invalid coordinates:", len(invalid_coords))
print("Percentage of invalid rows:", round(len(invalid_coords) / len(df_olist_geolocation) * 100, 4), "%")


Invalid coordinates: 25
Percentage of invalid rows: 0.0034 %


##### I decide to remove them, but I want to see where they are, bacasue I am curius :D

In [109]:
invalid_coords[['geolocation_zip_code_prefix', 'geolocation_lat', 'geolocation_lng', 'geolocation_city', 'geolocation_state']].head(10)


geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
387565                        18243        28.008978       -15.536867   
513631                        28165        41.614052        -8.411675   
513754                        28155        42.439286        13.820214   
514429                        28333        38.381672        -6.328200   
516682                        28595        43.684961        -7.411080   
538512                        29654        29.409252       -98.484121   
538557                        29654        21.657547      -101.466766   
585242                        35179        25.995203       -98.078544   
585260                        35179        25.995245       -98.078533   
695377                        45936        38.323939        -6.775035   

               geolocation_city geolocation_state  
387565  bom retiro da esperanca                sp  
513631      vila nova de campos                rj  
513754              santa maria                rj  
514429                   raposo                rj  
516682                  portela                rj  
538512   santo antônio do canaã                es  
538557   santo antonio do canaa                es  
585242       santana do paraíso                mg  
585260       santana do paraiso                mg  
695377                 itabatan                ba

##### This inspection of invalid rows revealed latitude and longitude values located far outside Brazil (e.g. Europe, North America, and Asia) despite Brazilian city names.
##### These were likely geocoding mismatches or input errors, so I remove them.

In [113]:
df_olist_geolocation = df_olist_geolocation.query(
    "-35 <= geolocation_lat <= 5 and -75 <= geolocation_lng <= -30"
)
print("✅ Removed 25 invalid coordinate rows.")
print("New shape:", df_olist_geolocation.shape)

✅ Removed 25 invalid coordinate rows.
New shape: (738307, 5)


### 4.3. Olist Order Items Dataset

In [117]:
df_olist_order_items.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14

In [121]:
df_olist_order_items.shape

(112650, 7)

In [123]:
df_olist_order_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB


In [125]:
df_olist_order_items.describe()

order_item_id          price  freight_value
count  112650.000000  112650.000000  112650.000000
mean        1.197834     120.653739      19.990320
std         0.705124     183.633928      15.806405
min         1.000000       0.850000       0.000000
25%         1.000000      39.900000      13.080000
50%         1.000000      74.990000      16.260000
75%         1.000000     134.900000      21.150000
max        21.000000    6735.000000     409.680000

### Observations from order_items dataset
#### Dataset Overview
##### 112,650 product records across 7 columns.
##### Each row represents one product within an order (one line item).

#### Expected columns:
##### order_id → unique ID of the order (links to orders dataset).
##### order_item_id → position of the product within the order.
##### product_id → product identifier.
##### seller_id → seller of the product.
##### shipping_limit_date → deadline for the seller to ship.
##### price → product price in BRL.
##### freight_value → shipping cost charged to the customer.

#### Data Quality Insights
##### No missing values — all columns have 112,650 non-null entries.
##### Data types are mostly correct:
##### Numeric columns (price, freight_value, order_item_id) → already numeric.
##### Date column (shipping_limit_date) → stored as object and needs conversion to datetime.
##### order_item_id ranges from 1 to 21 → some orders have multiple products.
##### Prices and freight values are realistic but vary widely (some high-value outliers).

#### Dataset is internally consistent and ready for analysis after small type adjustments.

#### Cleaning Steps:

In [129]:
df_olist_order_items.duplicated().sum()

0

In [133]:
# Convert shipping_limit_date from string (object) to proper datetime format
df_olist_order_items['shipping_limit_date'] = pd.to_datetime(df_olist_order_items['shipping_limit_date'])

In [137]:
# Verify data types again
df_olist_order_items.dtypes

order_id                       object
order_item_id                   int64
product_id                     object
seller_id                      object
shipping_limit_date    datetime64[ns]
price                         float64
freight_value                 float64
dtype: object

In [139]:
# Inspect numeric ranges for potential outliers
df_olist_order_items[['price', 'freight_value']].describe()

price  freight_value
count  112650.000000  112650.000000
mean      120.653739      19.990320
std       183.633928      15.806405
min         0.850000       0.000000
25%        39.900000      13.080000
50%        74.990000      16.260000
75%       134.900000      21.150000
max      6735.000000     409.680000

##### The numeric columns are healthy and reliable; the wide range simply reflects the diversity of products and shipping distances in the Olist marketplace.
##### Outliers are realistic high-end transactions, not data errors.

### 4.4. Olist Order Payments Dataset

In [144]:
df_olist_order_payments.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [148]:
df_olist_order_payments.shape

(103886, 5)

In [150]:
df_olist_order_payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


In [154]:
df_olist_order_payments.describe()

payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000

In [158]:
# Check duplicates
df_olist_order_payments.duplicated().sum()

# Check payment types
df_olist_order_payments['payment_type'].value_counts()

payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

In [166]:
# Check potential invalid (0) payments
df_olist_order_payments.query("payment_value == 0").shape[0]

9

##### I’m keeping the 9 zero-value payments because they’re rare (~0.009%), likely valid voucher or promo cases, and don’t affect data quality. They’ll be excluded only in revenue analyses if needed.

### Observations from order_payments dataset
#### Dataset Overview

##### Shape: (103,886, 5) → each row represents a single payment transaction.

##### Columns:
##### order_id → links payments to orders.
##### payment_sequential → sequence number for multiple payments on the same order.
##### payment_type → method used (credit_card, boleto, voucher, etc.).
##### payment_installments → number of installments chosen by the customer.
##### payment_value → total amount of that specific payment (in BRL).

#### Data Quality Insights
##### No missing values — all columns are fully populated.
##### Data types are correct (int, float, object).
##### Payment types are dominated by credit_card (~74%), followed by boleto, voucher, and debit_card.
##### Zero-value payments: only 9 rows (~0.009%), likely vouchers or promos — will be kept.

#### Numeric ranges are realistic:
##### payment_installments range 0–24 (mean ≈ 3).
##### payment_value range 0–13,664 BRL (wide spread but plausible for e-commerce).

#### Overall, the dataset is clean and consistent — ready for merging after minor validation.

### 4.5. Olist Order Reviews Dataset

In [173]:
df_olist_order_reviews.head()

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4  f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   

   review_score review_comment_title  \
0             4                  NaN   
1             5                  NaN   
2             5                  NaN   
3             5                  NaN   
4             5                  NaN   

                              review_comment_message review_creation_date  \
0                                                NaN  2018-01-18 00:00:00   
1                                                NaN  2018-03-10 00:00:00   
2                                                NaN  2018-02-17 00:00:00   
3              Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4  Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   

  review_answer_timestamp  
0     2018-01-18 21:46:59  
1     2018-03-11 03:05:13  
2     2018-02-18 14:36:24  
3     2017-04-21 22:02:06  
4     2018-03-02 10:26:53

In [177]:
df_olist_order_reviews.shape

(99224, 7)

In [181]:
df_olist_order_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   review_id                99224 non-null  object
 1   order_id                 99224 non-null  object
 2   review_score             99224 non-null  int64 
 3   review_comment_title     11568 non-null  object
 4   review_comment_message   40977 non-null  object
 5   review_creation_date     99224 non-null  object
 6   review_answer_timestamp  99224 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


In [185]:
df_olist_order_reviews.describe()

review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

In [187]:
# Check duplicates
df_olist_order_reviews.duplicated().sum()

0

In [199]:
#Convert date columns to datetime format
df_olist_order_reviews['review_creation_date'] = pd.to_datetime(df_olist_order_reviews['review_creation_date'])
df_olist_order_reviews['review_answer_timestamp'] = pd.to_datetime(df_olist_order_reviews['review_answer_timestamp'])

In [195]:
#Validate review scores
df_olist_order_reviews['review_score'].between(1,5).all()

True

### Observations from order_reviews dataset
#### Dataset Overview
##### Shape: (99,224, 7) → each row represents a single customer review for an order.

##### Columns:
##### review_id → unique identifier for each review.
##### order_id → connects review to the corresponding order.
##### review_score → rating from 1 to 5 stars.
##### review_comment_title → short title (optional).
##### review_comment_message → detailed comment (optional).
##### review_creation_date → date the review was created.
##### review_answer_timestamp → date/time when it was processed by the system.

#### Data Quality Insights
##### No missing values in key columns (review_id, order_id, review_score, and both dates).

#### Missing text fields:
##### review_comment_title: only about 12% of customers added a short title to their review — most skipped this optional field, which is common in online feedback.
##### review_comment_message: ~41% filled — common for short reviews.

#### Review scores: all within the valid range (1–5).
##### Mean = 4.09, Median = 5 → customers are generally satisfied.

#### Data types: dates are stored as object and need conversion to datetime.

##### Overall, the dataset is clean and consistent — perfect for merging with orders later to analyze satisfaction and delivery performance.

### 4.6. Olist Orders Dataset

In [204]:
df_olist_orders.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00

In [208]:
df_olist_orders.shape

(99441, 8)

In [212]:
df_olist_orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB


In [214]:
df_olist_orders.describe()

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522